flake.nix

```
@@ -91,6 +95,12 @@
               "mountroot d 777 0 0" # TODO permissions?
             ];
             extra-args = pkgs.lib.concatMapStrings (x: " -p \"${x}\"") extras;
+            exclude-args =
+              let
+                exclude-prefix = pkgs.lib.lists.optional (exclude != []) " -wildcards -e";
+                exclude-entries = map (x: " \"${x}\"") exclude;
+              in
+                pkgs.lib.strings.concatStrings (exclude-prefix ++ exclude-entries);
```

```suggestion
            exclude-args = pkgs.lib.optional (exclude != []) " -wildcards -e ${pkgs.lib.escapeShellArgs exclude}";
```

more robust escaping of exclude entries